File: VisionProApp/VisionProApp/Services/ContentService.swift
//
//  ContentService.swift
//  VisionProApp
//
//  Created by 五行 on 2025/5/11.
//
// Services/ContentService.swift
import Foundation

class ContentService: ObservableObject {
    @Published var contents: [Content] = []
    @Published var isLoading = false
    @Published var error: Error?
    
    // 获取内容列表
    func getContents(
        page: Int = 1,
        limit: Int = 20,
        status: String = "published",
        contentType: String? = nil,
        category: String? = nil
    ) async {
        isLoading = true
        error = nil
        
        do {
            contents = try await APIClient.shared.getContents(
                page: page,
                limit: limit,
                contentType: contentType,
                category: category
            )
        } catch {
            self.error = error
            print("获取内容列表失败: \(error)")
        }
        
        isLoading = false
    }
    
    // 创建内容
    func createContent(
        title: [String: String],
        description: [String: String]?,
        contentType: String,
        files: [String: Any],
        media: [String: Any]? = nil,
        location: [String: Any]? = nil,
        tags: [String]? = nil,
        category: String,
        pricing: [String: Any],
        isCollaborative: Bool = false
    ) async throws -> Content {
        isLoading = true
        error = nil
        
        do {
            let content = try await APIClient.shared.createContent(
                title: title,
                description: description,
                contentType: contentType,
                files: files,
                tags: tags,
                category: category,
                pricing: pricing
            )
            
            // 将新内容添加到列表
            contents.insert(content, at: 0)
            
            isLoading = false
            return content
        } catch {
            self.error = error
            isLoading = false
            throw error
        }
    }
    
    // 获取内容详情
    func getContent(contentId: String) async throws -> Content {
        let endpoint = "/contents/\(contentId)"
        let response: APIResponse<Content> = try await NetworkManager.shared.request(endpoint)
        
        guard let content = response.data else {
            throw APIError.serverError(response.message ?? "获取内容失败")
        }
        
        return content
    }
    
    // 获取用户内容
    func getUserContents(page: Int = 1, limit: Int = 20) async throws -> PaginatedResponse<Content> {
        let endpoint = "/contents/user"
        let queryParams = [
            "page": "\(page)",
            "limit": "\(limit)"
        ]
        
        let response: APIResponse<PaginatedResponse<Content>> = try await NetworkManager.shared.request(
            endpoint,
            queryParams: queryParams
        )
        
        guard let data = response.data else {
            throw APIError.serverError(response.message ?? "获取内容失败")
        }
        
        return data
    }
    
    // 更新内容
    func updateContent(contentId: String, updates: [String: Any]) async throws -> Content {
        let endpoint = "/contents/\(contentId)"
        let response: APIResponse<Content> = try await NetworkManager.shared.request(
            endpoint,
            method: .PUT,
            body: updates
        )
        
        guard let content = response.data else {
            throw APIError.serverError(response.message ?? "更新内容失败")
        }
        
        return content
    }
    
    // 删除内容
    func deleteContent(contentId: String) async throws {
        let endpoint = "/contents/\(contentId)"
        let _: APIResponse<EmptyResponse> = try await NetworkManager.shared.request(
            endpoint,
            method: .DELETE
        )
    }
}

File: vision-pro-frontend/src/services/api/contentService.js
// src/services/api/contentService.js
import { BaseApiService } from './base'

class ContentApiService extends BaseApiService {
  /**
   * 获取内容列表
   * @param {object} params - 查询参数
   * @returns {Promise<object>} - 内容列表和分页信息
   */
  async getContentList(params) {
    return this.get('/contents', params)
  }

  /**
   * 获取用户内容列表
   * @param {object} params - 查询参数
   * @returns {Promise<object>} - 用户内容列表和分页信息
   */
  async getUserContentList(params) {
    return this.get('/contents/user', params)
  }

  /**
   * 创建新内容
   * @param {object} data - 内容数据
   * @returns {Promise<object>} - 创建的内容
   */
  async createContent(data) {
    return this.post('/contents', data)
  }

  /**
   * 获取内容详情
   * @param {string} contentId - 内容ID
   * @returns {Promise<object>} - 内容详情
   */
  async getContentDetail(contentId) {
    return this.get(`/contents/${contentId}`)
  }

  /**
   * 更新内容
   * @param {string} contentId - 内容ID
   * @param {object} data - 更新数据
   * @returns {Promise<object>} - 更新后的内容
   */
  async updateContent(contentId, data) {
    return this.put(`/contents/${contentId}`, data)
  }

  /**
   * 删除内容
   * @param {string} contentId - 内容ID
   * @returns {Promise<object>} - 删除响应
   */
  async deleteContent(contentId) {
    return this.delete(`/contents/${contentId}`)
  }
  
  /**
   * 批量更新内容状态
   * @param {Array<string>} contentIds - 内容ID数组
   * @param {string} status - 目标状态
   * @returns {Promise<object>} - 更新响应
   */
  async batchUpdateStatus(contentIds, status) {
    return this.put('/creator/contents/batch/status', {
      contentIds,
      status
    })
  }
  
  /**
   * 批量添加标签
   * @param {Array<string>} contentIds - 内容ID数组
   * @param {Array<string>} tags - 标签数组
   * @returns {Promise<object>} - 更新响应
   */
  async batchAddTags(contentIds, tags) {
    return this.put('/creator/contents/batch/tags/add', {
      contentIds,
      tags
    })
  }
}

export const contentApiService = new ContentApiService()
